elements/pf-progress/pf-progress.ts

```diff
@@ -0,0 +1,189 @@
+ *          Background color of the progress bar.
+ *          {@default `#ffffff`}
+ *
+ * @cssprop {<color>} --pf-c-progress__status-icon--Color
+ *          Color of the status icon.
+ *          {@default `#151515`}
+ *
+ * @cssprop {<length>} --pf-c-progress__status-icon--MarginLeft
+ *          Margin left of the status icon.
+ *          {@default `0.5rem`}
+ *
+ * @cssprop {<length>} --pf-c-progress__indicator--Height
+ *          Height of the progress bar indicator.
+ *          {@default `1rem`}
+ *
+ * @cssprop {<color>} --pf-c-progress__indicator--BackgroundColor
+ *          Background color of the progress bar indicator.
+ *          {@default `#ffffff`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-success__bar--BackgroundColor
+ *          Background color of the progress bar when variant is success.
+ *          {@default `#3e8635`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-warning__bar--BackgroundColor
+ *          Background color of the progress bar when variant is warning.
+ *          {@default `#f0ab00`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-danger__bar--BackgroundColor
+ *          Background color of the progress bar when variant is danger.
+ *          {@default `#c9190b`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-success__status-icon--Color
+ *          Color of the status icon when variant is success.
+ *          {@default `#3e8635`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-warning__status-icon--Color
+ *          Color of the status icon when variant is warning.
+ *          {@default `#f0ab00`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-danger__status-icon--Color
+ *          Color of the status icon when variant is danger.
+ *          {@default `#c9190b`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-success--m-inside__measure--Color
+ *          Color of the progress bar measure when variant is success and measure location is inside.
+ *          {@default `#ffffff`}
+ *
+ * @cssprop {<length>} --pf-c-progress--m-outside__measure--FontSize
+ *          Font size of the progress bar measure when measure location is outside.
+ *          {@default `0.875rem`}
+ *
+ * @cssprop {<length>} --pf-c-progress--m-sm__bar--Height
+ *          Height of the progress bar when the size is small.
+ *          {@default `0.5rem`}
+ *
+ * @cssprop {<length>} --pf-c-progress--m-sm__description--FontSize
+ *          Font size of the progress bar description when the size is small.
+ *          {@default `0.875rem`}
+ *
+ * @cssprop {<length>} --pf-c-progress--m-lg__bar--Height
+ *          Height of the progress bar when the size is large.
+ *          {@default `1.5rem`}
+ *
+ */
+@customElement('pf-progress')
+export class PfProgress extends LitElement {
+  static readonly styles = [styles];
+
+  static readonly formAssociated = true;
+
+  /** Represents the value of the progress bar */
+  @property({ reflect: true, type: Number })
+    value = 0;
```

Comment:
since this is formAssociated, we'll need to setValue on the internals object when the value changes. Please add a test for form submission

Reply:
Trying to think through whether we need it to be form associated anymore since we are not having the progress bar be accessible.  It would still have a value but be more like an image than an actual input.

Reply:
We'd still want the host element to submit it's value with the form though. Setting the value property on the host object isn't enough, we need to call setValue on the internals object or the formdata won't read the progress value

Reply:
Discussed today during office hours, the native `progress` element itself does not report its value to the form element.  Further, it appears that forms only submit a control that is outlined in the spec https://www.w3.org/TR/html401/interact/forms.html#form-controls which can be a input, text area, menu, and/or `hidden` | `generic` options set through `input` | `object` respectively. As such, we will not include form association for progress element as this time.